h5pt.go

```diff
@@ -108,11 +119,27 @@ func (t *Table) ReadPackets(start, nrecords int, data interface{}) error {
 // Struct values must only have exported fields, otherwise extractValues will
 // panic.
 func extractValues(ad *appendData, data interface{}) error {
+	if data, ok := data.(interface{ CMarshal() ([]byte, error) }); ok {
```

Comment:
It's nicer to name interfaces that get used like this, just for the sake of readability.

Reply:
Of course! WIP :)